nodebuilder/tests/prune_test.go

```diff
@@ -0,0 +1,152 @@
+package tests
+
+import (
+	"bytes"
+	"context"
+	"testing"
+	"time"
+
+	"github.com/libp2p/go-libp2p/core/host"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/fx"
+
+	"github.com/celestiaorg/celestia-node/blob"
+	"github.com/celestiaorg/celestia-node/nodebuilder"
+	"github.com/celestiaorg/celestia-node/nodebuilder/das"
+	"github.com/celestiaorg/celestia-node/nodebuilder/node"
+	"github.com/celestiaorg/celestia-node/nodebuilder/tests/swamp"
+	"github.com/celestiaorg/celestia-node/pruner"
+	"github.com/celestiaorg/celestia-node/share"
+)
+
+// 1 BN w/ pruning
+// 3 FN w/ pruning
+// 1 FN archival
+
+// turn on archival BN
+// archival FN syncs against BN
+// turn off archival BN
+// turn on pruning BN
+// spin up 3 pruning FNs, connect
```

Comment:
Based on the description, it looks like those cases are covered:
- Sync from archival_BN to archival_FN
- Sync from archival_BN+pruning_BN to pruning_FN
- LN sampling from archival_FN

What is the purpose of the second test? It doesn’t seem to be testing anything meaningful. Perhaps it should be broken into separate test cases to isolate routing conditions:

- Pruning_FN can sync from only pruning_FN/BN.
- Pruning_FN from only archival_FN.

Also missing:
- Archival_FN from archival_FN. There is a test syncing from archival_BN, which is similar but not the same.

In the current setup, LN requests from both archival and pruning run at the same time. I would also suggest breaking the test into two:
- LN can manually request archival and recent blobs from ONLY archival_FN.
- LN can manually request recent blobs from ONLY pruning_FN.

Reply:
> What is the purpose of the second test? It doesn’t seem to be testing anything meaningful. 

which second test do you mean? There is only one test here that tests to ensure a LN can discover + sync historic blobs off of an archival node in a network topology where pruned FNs are dominant.

> LN can manually request recent blobs from ONLY pruning_FN

We can't control that the LN requests recent blobs only from pruned FNs as archival nodes can serve the exact same data and also advertise as fulls in the network.

Reply:
I agree we should add an additional test case for a pruned FN requesting historic data from archival FN/BN though, but I'd like to put that in an issue and address it after this PR.

Reply:
>which second test do you mean? There is only one test here that tests to ensure a LN can discover + sync historic blobs off of an archival node in a network topology where pruned FNs are dominant.

Sorry for misleading wording, I meant the second case I've listened in covered cases. I. referred to  it as second test, because single test covers multiples scenarios.

>We can't control that the LN requests recent blobs only from pruned FNs as archival nodes can serve the exact same data and also advertise as fulls in the network.

Imagine pruned are not available and it always falls back to archival serving. We need the test to ensure pruned can serve too. You can shutdown archival to ensure pruned are able to serve.